### PR TITLE
Add support for Helm v3 dependency management

### PR DESCRIPTION
**Pull request by tompizmor**
Since Helm v3 the recommended way to specify chart dependencies is directly in the Chart.yaml file and Chart.lock instead of requirements.yaml and requirements.lock. 

This PR adds support for both kinds of dependency management by checking the existence of `requirements.lock` or `Chart.lock` in the processed chart.

### CODE REVIEW

File: pkg/chart/dependency.go [hunk 1]
```diff
@@ -22,20 +22,28 @@ type dependencies struct {
...
-	requirementsLockFile := path.Join(chartPath, "requirements.lock")
+	lockFileName := ""
+	if apiVersion == "v1" {
+		lockFileName = "requirements.lock"
+	} else if apiVersion == "v2" {
+		lockFileName = "Chart.lock"
+	} else {
+		return errors.Errorf("unrecognised apiVersion %s", apiVersion)
+	}
+	lockFile := path.Join(chartPath, lockFileName)
```
Comment (jotadrilo) [inline pkg/chart/dependency.go:37]: I would add a small function for this (and a test separately)

```
func lockFilePath(chartPath, apiVersion) (string, error) {
}
```

```
err, lockFile := lockFilePath(chartPath, apiVersion)
if err != nil { ... }
```

File: pkg/chart/dependency.go [hunk 2]
```diff
@@ -22,20 +22,28 @@ type dependencies struct {
...
 	var missingDependencies = false
-	requirementsLockFile := path.Join(chartPath, "requirements.lock")
+	lockFileName := ""
+	if apiVersion == "v1" {
+		lockFileName = "requirements.lock"
+	} else if apiVersion == "v2" {
+		lockFileName = "Chart.lock"
+	} else {
+		return errors.Errorf("unrecognised apiVersion %s", apiVersion)
+	}
```
Comment (jotadrilo) [inline pkg/chart/dependency.go:36]: please be consistent with the usage of `if` or `switch`

File: pkg/chart/dependency.go [hunk 3]
```diff
@@ -77,9 +85,18 @@ func syncDependencies(chartPath string, sourceRepo *api.Repo, target *api.Target
 	}
 
 	if !missingDependencies {
-		klog.V(3).Info("Updating requirements.yaml file...")
-		if err := updateRequirementsFile(chartPath, lock, sourceRepo, target); err != nil {
-			return errors.Trace(err)
+		klog.V(3).Info("Updating Chart.yaml file...")
+		switch apiVersion {
+		case "v1":
```
Comment (jotadrilo) [inline pkg/chart/dependency.go:90]: I would add a `types.go` file stating this kind of constants:

```golang
package chart

const (
  API_V1 string = "v1"
  API_V2 string = "v2"
)
```

you can use this file to define common constants such as "Chart.yaml"

File: pkg/chart/dependency.go (hunk 1 above)
Comment (tompizmor) [reply pkg/chart/dependency.go]: done!

File: pkg/chart/dependency_test.go [hunk 4]
```diff
@@ -163,3 +221,51 @@ func TestFindDepByName(t *testing.T) {
...
+			errors.New("unrecognised apiVersion \"vvv000\""),
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			want := tc.expectedPath
+			got, err := lockFilePath(tc.chartPath, tc.apiVersion)
+			if tc.shouldFail {
+				if !assert.EqualError(t, tc.expectedError, err.Error()) {
```
Comment (jotadrilo) [inline pkg/chart/dependency_test.go:261]: why do you need `assert`?

File: pkg/chart/dependency.go [hunk 5]
```diff
@@ -77,9 +80,18 @@ func syncDependencies(chartPath string, sourceRepo *api.Repo, target *api.Target
 	}
 
 	if !missingDependencies {
-		klog.V(3).Info("Updating requirements.yaml file...")
-		if err := updateRequirementsFile(chartPath, lock, sourceRepo, target); err != nil {
-			return errors.Trace(err)
+		klog.V(3).Info("Updating Chart.yaml file...")
```
Comment (jotadrilo) [inline pkg/chart/dependency.go:83]: I would just say "Updating dependencies file..."

File: pkg/chart/dependency.go [hunk 6]
```diff
@@ -88,9 +100,39 @@ func syncDependencies(chartPath string, sourceRepo *api.Repo, target *api.Target
...
+func updateChartMetadataFile(chartPath string, lock *helmChart.Lock, sourceRepo *api.Repo, target *api.TargetRepo) error {
+	chartFile := path.Join(chartPath, ChartFilename)
+	chart, err := ioutil.ReadFile(chartFile)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	chartMetadata := &helmChart.Metadata{}
+	err = yaml.Unmarshal(chart, chartMetadata)
+	if err != nil {
+		return errors.Annotatef(err, "Error unmarshaling %s file", chartFile)
```
Comment (jotadrilo) [inline pkg/chart/dependency.go:114]: remember that errors should not be capitalized: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

**Review by jotadrilo — APPROVED** [review]
LGTM!

File: pkg/chart/dependency_test.go (hunk 4 above)
Comment (tompizmor) [reply pkg/chart/dependency_test.go:261]: I don't really need it. I can compare errors strings:

~~~go
err.Error() != tc.expectedError.Error() {
...
}
~~~

File: pkg/chart/dependency.go (hunk 6 above)
Comment (tompizmor) [reply pkg/chart/dependency.go:114]: Just for you to know, the new errors I wrote in this PR I did it in lower case, but this one comes from a copy-paste.

Hopefully, you don't have to remind me this rule anymore 😄

**Pull request merged**